x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/entsearch/engine/action/PutEngineAction.java

```diff
@@ -0,0 +1,129 @@
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.entsearch.engine.action;
+
+import org.elasticsearch.action.ActionRequest;
+import org.elasticsearch.action.ActionRequestValidationException;
+import org.elasticsearch.action.ActionResponse;
+import org.elasticsearch.action.ActionType;
+import org.elasticsearch.action.DocWriteResponse;
+import org.elasticsearch.action.IndicesRequest;
+import org.elasticsearch.action.support.IndicesOptions;
+import org.elasticsearch.common.bytes.BytesReference;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.xcontent.StatusToXContentObject;
+import org.elasticsearch.rest.RestStatus;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xcontent.XContentType;
+import org.elasticsearch.xpack.entsearch.engine.Engine;
+
+import java.io.IOException;
+
+import static org.elasticsearch.action.ValidateActions.addValidationError;
+
+public class PutEngineAction extends ActionType<PutEngineAction.Response> {
+
+    public static final PutEngineAction INSTANCE = new PutEngineAction();
+    public static final String NAME = "indices:admin/engine/put";
+
+    public PutEngineAction() {
+        super(NAME, PutEngineAction.Response::new);
+    }
+
+    public static class Request extends ActionRequest implements IndicesRequest.Replaceable {
+
+        private final Engine engine;
+
+        public Request(StreamInput in) throws IOException {
+            super(in);
+            this.engine = new Engine(in);
+        }
+
+        public Request(String engineId, BytesReference content, XContentType contentType) {
+            this.engine = Engine.fromXContentBytes(engineId, content, contentType);
+        }
+
+        public Request(Engine engine) {
+            this.engine = engine;
+        }
+
+        @Override
+        public ActionRequestValidationException validate() {
+            ActionRequestValidationException validationException = null;
+
+            if (engine.indices().length == 0) {
+                validationException = addValidationError("indices are missing", validationException);
+            }
+
+            return validationException;
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            super.writeTo(out);
+            engine.writeTo(out);
+        }
+
+        @Override
+        public String[] indices() {
+            return engine.indices();
+        }
+
+        @Override
+        public IndicesRequest indices(String... indices) {
+            return new Request(new Engine(engine.name(), indices, engine.analyticsCollectionName()));
+        }
+
+        @Override
+        public IndicesOptions indicesOptions() {
+            return IndicesOptions.fromOptions(false, false, false, false, false, false, false, false);
```

Comment:
I don't think we enable the right options here. We should copy what is done for aliases [here](https://github.com/elastic/elasticsearch/blob/main/server/src/main/java/org/elasticsearch/action/admin/indices/alias/IndicesAliasesRequest.java#L60).

Reply:
I'm fine with that. Do we want to allow wildcard expansions? Would it be possible then to issue the following request?

PUT /_engine/test-engine-another-one
{
  "indices": ["test*"]
}

Reply:
Was it allowed in the rails app? If not we can just disable index pattern resolution in the IndexOptions.

Reply:
This was not allowed in the Rails app. I'll disable index pattern resolution.